src/v/redpanda/debug_bundle.cc

```diff
@@ -0,0 +1,266 @@
+    auto last_slash_pos = rpk_path.find_last_of('/');
+    if (last_slash_pos == ss::sstring::npos) {
+        throw ss::httpd::server_error_exception(
+          fmt::format("Invalid RPK path: {}", rpk_path));
+    }
+
+    ss::sstring last_field = rpk_path.substr(last_slash_pos + 1);
+
+    return last_field == "rpk" ? rpk_path : rpk_path + "/rpk";
+}
+
+debug_bundle::debug_bundle(ss::sstring filename, ss::sstring rpk_path)
+  : _status{debug_bundle_status::initialized}
+  , _filename{filename}
+  , _rpk_cmd{make_rpk_cmd(rpk_path)}
+  , _transfer_handle{_filename, nullptr, false} {}
+
+ss::future<> debug_bundle::stop() {
+    vlog(bundle_log.info, "Stopping debug bundle ...");
+    return ss::make_ready_future<>();
+}
+
+ss::future<> debug_bundle::start_creating_bundle(
+  const request_auth_result& auth_state, const debug_bundle_params params) {
+    if (ss::this_shard_id() != debug_bundle_shard_id) {
+        return container().invoke_on(
+          debug_bundle_shard_id,
+          [&auth_state, params{std::move(params)}](debug_bundle& b) {
+              return b.start_creating_bundle(auth_state, std::move(params));
+          });
+    }
+
+    ss::sstring host_path;
+    auto host_env = std::getenv("PATH");
+    if (!host_env) {
+        vlog(
+          bundle_log.warn,
+          "Failed to get 'PATH' environmental variable, the debug bundle may "
+          "be incomplete due to missing dependencies");
+    } else {
+        vlog(bundle_log.trace, "Host path: {}", host_env);
+        host_path = fmt::format("PATH={}", host_env);
+    }
+
+    std::vector<ss::sstring> rpk_argv{
+      _rpk_cmd, "debug", "bundle", "--output", _filename};
+    // Add SASL creds to RPK flags if SASL is enabled.
+    // No need to check for sasl on the broker endpoint because that is for
+    // Kafka brokers where as the bundle is managed by the Admin server only.
+    if (config::shard_local_cfg().enable_sasl()) {
+        rpk_argv.push_back("--user");
+        rpk_argv.push_back(auth_state.get_username());
+        rpk_argv.push_back("--password");
+        rpk_argv.push_back(auth_state.get_password());
+        rpk_argv.push_back("--sasl-mechanism");
+        rpk_argv.push_back(auth_state.get_mechanism());
+    }
+
+    if (params.logs_since.has_value()) {
+        rpk_argv.push_back("--logs-since");
+        rpk_argv.push_back(params.logs_since.value());
+    }
+
+    if (params.logs_until.has_value()) {
+        rpk_argv.push_back("--logs-until");
+        rpk_argv.push_back(params.logs_until.value());
+    }
+
+    if (params.logs_size_limit.has_value()) {
+        rpk_argv.push_back("--logs-size-limit");
+        rpk_argv.push_back(params.logs_size_limit.value());
+    }
+
+    if (params.metrics_interval.has_value()) {
+        rpk_argv.push_back("--metrics-interval");
+        rpk_argv.push_back(params.metrics_interval.value());
+    }
+
+    vlog(
+      bundle_log.trace, "RPK absolute path {}, flags {}", _rpk_cmd, rpk_argv);
+    ssx::background
```

Comment:
not seeing a gate anywhere. am i missing it?

Reply:
nah, I just didn't write it in yet.